numbagg/grouped.py

```diff
@@ -1,11 +1,32 @@
 import numpy as np
+from numba.types import float32, float64, int32, int64
 
 from .decorators import groupndreduce
 
-
-@groupndreduce.wrap()
-def group_nanmean(values, labels, out):
-    counts = np.zeros(out.shape, dtype=labels.dtype)
+dtypes = [
+    (int32, int32, int32),
+    (int32, int64, int32),
+    (int64, int32, int64),
+    (int64, int64, int64),
+    (float32, int32, float32),
+    (float32, int64, float32),
+    (float64, int32, float64),
+    (float64, int64, float64),
+]
+dtypes_counts = [
+    (int32, int32, int64, int32),
+    (int32, int64, int64, int32),
+    (int64, int32, int64, int64),
+    (int64, int64, int64, int64),
+    (float32, int32, int64, float32),
+    (float32, int64, int64, float32),
+    (float64, int32, int64, float64),
+    (float64, int64, int64, float64),
+]
```

Comment:
We've since changed how we specify the signature to include `int8`, so I'll take a pass at updating this. In particular, we should update the `int8` tests to include a function that uses the counts signature.